optimum/onnxruntime/modeling_decoder.py

```diff
@@ -465,121 +484,214 @@ def _from_pretrained(
         config: "PretrainedConfig",
         use_auth_token: Optional[Union[bool, str]] = None,
         revision: Optional[str] = None,
-        force_download: bool = True,
+        force_download: bool = False,
         cache_dir: Optional[str] = None,
         decoder_file_name: str = ONNX_DECODER_NAME,
         decoder_with_past_file_name: str = ONNX_DECODER_WITH_PAST_NAME,
         subfolder: str = "",
         local_files_only: bool = False,
         use_cache: bool = True,
-        use_io_binding: bool = True,
         provider: str = "CPUExecutionProvider",
         session_options: Optional[onnxruntime.SessionOptions] = None,
         provider_options: Optional[Dict[str, Any]] = None,
+        use_io_binding: bool = True,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
     ):
-        file_names = {}
-        # Load model from a local directory
-        if os.path.isdir(os.path.join(model_id, subfolder)):
-            decoder_with_past_path = (
-                os.path.join(model_id, subfolder, decoder_with_past_file_name) if use_cache else None
+        model_path = Path(model_id)
+
+        def validate_filename(filename):
+            if model_path.is_dir():
+                return (model_path / subfolder / filename).is_file()
+            succeeded = True
+            try:
+                get_hf_file_metadata(hf_hub_url(model_id, filename, subfolder=subfolder, revision=revision))
+            except Exception:
+                succeeded = False
+            return succeeded
+
+        def infer_filename(pattern: str, argument_name: str, fail_if_not_found: bool = True) -> str:
```

Comment:
Same as above (Can we add a short comment as to what this function does + wouldn't it be better out)